ouroboros-consensus/src/Ouroboros/Consensus/NodeKernel.hs

```diff
@@ -175,7 +186,20 @@ initNodeKernel args@NodeKernelArgs { registry, cfg, tracers
       , getFetchMode           = readFetchMode blockFetchInterface
       , getNodeCandidates      = varCandidates
       , getTracers             = tracers
+      , setBlockForging        = \a -> atomically . LazySTM.writeTVar blockForgingVar $! a
       }
+  where
+    blockForgingController :: InternalState m remotePeer localPeer blk
+                           -> STM m [BlockForging m blk]
+                           -> m Void
+    blockForgingController st getBlockForging = go []
+      where
+        go :: [Thread m Void] -> m Void 
```

Comment:
As written, this is a busy loop, no?

I think this may be a good use case for https://github.com/input-output-hk/ouroboros-network/blob/327a90918caabaa3d5fadb54de2701ca4fc7a83f/ouroboros-consensus/src/Ouroboros/Consensus/Util/STM.hs#L158

Or perhaps this should be a `TMVar`? And thus (at most) one cycling of the threads per `setBlockForging`

Reply:
Fixed.

Reply:
I haven't used `Watcher` because it seemed to me to be designed for a more complex usecase.  We don't need a finger print, and we just need to block on the `TMVar` until it changes.  However, in the future, if the `TMVar` holds configuration from which `BlockForging` record is constructed, then a `Watcher` will actually make sense.